detector.go

```diff
@@ -60,36 +60,43 @@ type Detector struct {
 }
 
 func (f *DetectorFactory) NewDetector(appDir, orderPath, platformDir string, logger log.Logger) (*Detector, error) {
-	orderBp, orderExt, err := f.configHandler.ReadOrder(orderPath)
-	if err != nil {
-		return nil, err
-	}
-	if f.platformAPI.LessThan("0.10") {
-		orderExt = nil
-	}
-	if err = f.verifyAPIs(orderBp, orderExt); err != nil {
-		return nil, err
-	}
-
-	return &Detector{
+	detector := &Detector{
 		AppDir:      appDir,
 		DirStore:    f.dirStore,
 		Logger:      logger,
-		Order:       PrependExtensions(orderBp, orderExt),
 		PlatformDir: platformDir,
 		Resolver:    &DefaultResolver{Logger: logger},
 		Runs:        &sync.Map{},
-	}, nil
+	}
+	if err := f.setOrder(detector, orderPath, logger); err != nil {
+		return nil, err
+	}
+	return detector, nil
+}
+
+func (f *DetectorFactory) setOrder(detector *Detector, path string, logger log.Logger) error {
+	orderBp, orderExt, err := f.configHandler.ReadOrder(path)
+	if err != nil {
+		return err
+	}
+	if f.platformAPI.LessThan("0.10") {
+		orderExt = nil
+	}
```

Comment:
```suggestion
```
We should let the `main` package decide whether to proceed if there are extensions